src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
>  I do think we need to make buffer returns 100% reliable in our code

I agree on 99% ;)  But I'm not sure that perfectly returning all the buffers all the time is worth the code complexity and potentially perf impact on the common/success path.  If you look at StringBuilderCache in mscorlib, for example, usage of it doesn't spend any effort to try to put back the builder on error paths... if an exception is thrown and the builder isn't put back, a new one is simply created the next time around.  That'll happen as well with ArrayPool, and with the infrequency and relative cost of exceptions, plus the code complexity added by using try/finallies, and potentially the perf impact from doing so (even minor things like functions not being inlinable where they were before by now having try/finally) or in this case by, gulp, adding finalizers, I really question whether it's worth it to achieve 100%.